client/src/views/UserProfile.tsx

```diff
@@ -69,24 +72,65 @@ export function UserProfile() {
   const { id } = useParams()
   const isCurrentUser = id === currentUser?.discordId
   const [isEdit, setIsEdit] = useState(false)
-  const [successOpen, setSuccessOpen] = useState(false)
-  const [failureOpen, setFailureOpen] = useState(false)
+  const [snackBarOpen, setSnackBarOpen] = useState(false)
+  const [requestError, setRequestError] = useState(false)
+  const [snackBarMessage, setSnackBarMessage] = useState('')
+  const [canToggle, setCanToggle] = useState(true)
 
   const [user, setUser, error, isLoading] = useUser(id)
+  const [jigokuName, setJigokuName] = useState()
+  const [preferredClan, setPreferredClan] = useState()
```

Comment:
**Improvement:** When you get a lot of `useState` like this, it's usually easier to handle and better for performance to use `useReducer`